lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
Regarding the .NET version: I believe we currently *target* 4.5. That is independent of what you have on your machine.

Also, it seems we are really dealing with two issues:

  1. IMRU uses too much memory and is inefficient in its object (de-)allocations. Some object reuse as suggested by @anupam128 should help. 
  2. The memory constraint of a container isn't enforced in a way that the CLR knows about. This means that GC isn't invoked when it should. Calling the GC on our end won't really solve that issue, just delay it.

Can we focus on the 1st problem in this issue / PR and create a JIRA for the second one?